### ISSUE TITLE
[BUG] - DatePicker's showMonthAndYearPickers mode changes the calendar type on scrolling the year picker

### ISSUE THREAD

**Issue by Hasan-Mir** [nextui-org/nextui#3059]
### NextUI Version

2.3.6

### Describe the bug

When using a locale for the `DatePicker` component with the `showMonthAndYearPickers` enabled, the calendar locale changes while scrolling the **years** list.

**locale="fa-IR"**:

```js
import { DatePicker } from "@nextui-org/react";
import { I18nProvider } from "@react-aria/i18n";
import "./App.css";

function App() {
  return (
    <div>
      <I18nProvider locale="fa-IR">
        <DatePicker showMonthAndYearPickers />
      </I18nProvider>
    </div>
  );
}

```

https://github.com/nextui-org/nextui/assets/62181905/5a8d3fa7-cf3c-4b7a-affc-d0f4159e5dfc

**locale="hi-IN-u-ca-indian"**:

```js
function App() {
  return (
    <div>
      <I18nProvider locale="hi-IN-u-ca-indian">
        <DatePicker showMonthAndYearPickers />
      </I18nProvider>
    </div>
  );
}
```

https://github.com/nextui-org/nextui/assets/62181905/f6e85369-10dd-4293-9722-352c474aa188

### Your Example Website or App

https://codesandbox.io/p/devbox/friendly-leftpad-g56gcc?file=%2FApp.tsx&utm_medium=sandpack

### Steps to Reproduce the Bug or Issue

1. Create a DatePicker and provide `locale` and `showMonthAndYearPickers` props.
2. Open the month and year pickers and scroll the **years** list.

### Expected behavior

It should not change the name of the months and the year from the selected locale to Gregorian.

### Operating System Version

Windows 10

### Browser

Chrome

**Comment by MHBahrampour** [comment]
Almost two month later using version 2.4.2 and the issue still persists.

**Comment by wingkwong** [comment]
@MHBahrampour The fix will be available in the upcoming release v2.4.3.

**Comment by MHBahrampour** [comment]
@wingkwong Thanks. I really need it for my new project.

**Comment by abhirakshit** [comment]
Following!

**Comment by wingkwong** [comment]
Closing - fixed in above PR. The fix will be available in the upcoming bug fix release v2.4.3.
